test/appium/tests/atomic/chats/test_public.py

```diff
@@ -328,6 +329,304 @@ def test_public_chat_open_using_deep_link(self):
+            self.errors.append("Counter of reaction is not set on Emoji for message receiver!")
+        self.chat_1.set_reaction(emoji_unicode, emoji_message=True)
+        if emoji_message.emojis_below_message(own=False) == 1:
+            self.errors.append("Counter of reaction is not re-set on Emoji for message receiver!")
+
+        self.errors.verify_no_errors()
+
+    @marks.testrail_id(702067)
+    @marks.medium
+    def test_messaging_open_user_profile_long_press_on_message_1_1_chat(self):
+
+        message_from_sender = "Message sender"
+
+        self.home_1.just_fyi("1-1 chat: sender adds receiver and send a message")
+        self.chat_1.send_message(message_from_sender)
+        self.chat_1.chat_element_by_text(message_from_sender).long_press_element()
+        if self.chat_1.view_profile_by_avatar_button.is_element_displayed():
+            self.errors.append('Member photo is shown on long tap on sent message from 1-1 chat')
+        self.chat_1.click_system_back_button(2)
+
+        self.home_2.just_fyi("1-1 chat: receiver verifies that can open sender profile on long tap on message")
+        self.chat_2.chat_element_by_text(message_from_sender).long_press_element()
+        if self.chat_2.view_profile_by_avatar_button.is_element_displayed():
+            self.errors.append('1-1 chat: another user profile is opened on long tap on received message')
+        self.chat_2.click_system_back_button(2)
+
+    @marks.testrail_id(702068)
+    @marks.medium
+    def test_messaging_open_user_profile_long_press_on_message_in_public_chat(self):
+        message_from_receiver = "Message receiver"
+        # [chat.home_button.double_click() for chat in [self.chat_1, self.chat_2]]
+        [home.get_chat('#' + self.public_chat_name).click() for home in (self.home_1, self.home_2)]
+        self.home_1.just_fyi(
+            'Public chat: send message and verify that user profile can be opened on long press on message')
+        # chat_public_1, chat_public_2 = home_1.get_chat_view(), home_2.get_chat_view()
+        self.public_chat_2.send_message(message_from_receiver)
+        self.public_chat_2.chat_element_by_text(message_from_receiver).long_press_element()
+        if self.public_chat_2.view_profile_by_avatar_button.is_element_displayed():
+            self.errors.append('Public chat: "view profile" is shown on long tap on sent message')
+        self.public_chat_1.view_profile_long_press(message_from_receiver)
+        if not self.public_chat_1.remove_from_contacts.is_element_displayed():
+            self.errors.append('Public chat: another user profile is not opened on long tap on received message')
+
+        self.errors.verify_no_errors()
+
+        self.public_chat_1.get_back_to_home_view()
+        self.public_chat_2.click_system_back_button(2)
+
+        # [chat.home_button.double_click() for chat in [self.public_chat_1, self.public_chat_2]]
+        self.home_1.get_chat(self.default_username_2).click()
```

Comment:
navigation for next test must be in next test; so if you are not on the desired view when start `test_messaging_can_pin_messages_in_one_to_one_chat` - better to navigate home > navigate to chat view

Reply:
I have a question here: if we run test_messaging_can_pin_messages_in_one_to_one_chat separately, then it does not make sense to navigate to chat view, as we already navigated there after setup class. 

Navigating to chat view should be performed only in case if test_messaging_can_pin_messages_in_one_to_one_chat is running after test_messaging_open_user_profile_long_press_on_message_in_public_chat that's why I decided to set navigation for next test in previous one. 

Could you please explain, what is the reason of why we need to keep navigation for the next test exactly in next test?

Reply:
Yes, sure.
Let's say the previous test failed somewhere in the middle.
From that point, this particular test will be failed just because of the reason that previous test couldn't finish